### PR TITLE
Matchmaking is working - needs some light cleanup

### PR DESCRIPTION
**Pull request by rickpr**

### CODE REVIEW

File: components/admin/db/seeds.rb [hunk 1]
```diff
@@ -36,8 +36,9 @@ def initialize_taxonomies!
       'Skills to Learn',
       'Skills Needed',
     ].map(&:to_sym)
-    taxonomy_names.each do |taxonomy_name|
-      Taxonomy.find_or_create_by!(name: taxonomy_name)
+    taxonomy_weights = [1, 2, 2]
+    taxonomy_names.zip(taxonomy_weights).each do |taxonomy_name, weight|
+      Taxonomy.find_or_create_by!(name: taxonomy_name, weight: weight)
```
Comment (rickpr) [inline components/admin/db/seeds.rb]: This is unnecessary and should probably be built as an attribute hash, then the names extracted from it.

File: components/admin/spec/controllers/matches_controller_spec.rb [hunk 2]
```diff
@@ -0,0 +1,12 @@
+require 'rails_helper'
+
+RSpec.describe MatchesController, type: :controller do
+
+  let(:valid_attributes) { { taxonomies: { 1 => [{'tagId'=>1}] } } }
+
+  describe 'POST #create' do
+    it 'creates a new Match' do
+      pending 'Need to seed the test db'
```
Comment (rickpr) [inline components/admin/spec/controllers/matches_controller_spec.rb]: This should come next - a seeded test db could yield some easy but useful tests.

I wanted to capture at least the parameter format for now.

File: components/admin/spec/routing/matches_routing_spec.rb [hunk 3]
```diff
@@ -0,0 +1,9 @@
+require 'rails_helper'
+
+RSpec.describe MatchesController, type: :routing do
+  describe 'routing' do
+    it 'routes to #create' do
+      expect(post: '/matches').to route_to('matches#create')
```
Comment (rickpr) [inline components/admin/spec/routing/matches_routing_spec.rb]: This spec is potentially dumb and might be removable.

File: components/react/src/App.js [hunk 4]
```diff
@@ -1,56 +1,52 @@
...
+  const [error, setError] = useState();
+  const [selected, setSelected] = useState({});
+  const [matchScores, setMatchScores] = useState({});
+
+  useEffect(
+    () => request('projects', setProjects).then(() => request('taxonomies', setTaxonomies)).then(() => setIsLoaded(true)),
+    [],
+  );
+  const request = (endpoint, setter) => {
+    return fetch(`http://localhost:5455/${endpoint}.json`, { signal: abortController.signal })
```
Comment (rickpr) [inline components/react/src/App.js]: According to React, we should use `async/await` instead.

File: components/react/src/Components/Selector/Selector.js [hunk 5]
```diff
@@ -1,28 +1,33 @@
...
 import _ from 'lodash';
 
-const Selector = ({ categories, name, taxonomyId }) => {
+const Selector = ({ categories, name, taxonomyId, addToMediator }) => {
   const [selected, setSelected] = useState([]);
+  useEffect(() => addToMediator(selected.map(getTagId)));
 
   const selectorCategory = (categoryName, id) =>
-    <Category categoryId={id} name={categoryName} click={() => removeCategoryFromSelected(id)} key={'category' + id} />;
+    <Category categoryId={id} name={categoryName} click={() => removeCategoryFromSelected(id)} key={`category${id}`} />;
```
Comment (rickpr) [inline components/react/src/Components/Selector/Selector.js]: This `categoryId` vs `tagId` thing is lame. How can we improve it? It makes this weird and matchmaking kind of hard.

File: components/admin/app/helpers/matches_helper.rb [hunk 6]
```diff
@@ -0,0 +1,2 @@
+module MatchesHelper
+end
```
Comment (rickpr) [inline components/admin/app/helpers/matches_helper.rb]: This isn't used and should be deleted (YAGNI).

File: components/admin/app/models/match.rb [hunk 7]
```diff
@@ -0,0 +1,34 @@
...
+  end
+
+  def match_score(project)
+    taxonomies.sum do |key, tags_and_categories|
+      category_ids = tags_and_categories[:categories]
+      tag_ids = tags_and_categories[:tags]
+      weight = Taxonomy.find(key).weight
+      matched_categories = project.categories.where(id: category_ids)
+      matched_tags = project.tags.where(id: tag_ids)
+      sum = (matched_categories.count + matched_tags.count) * weight
```
Comment (rickpr) [inline components/admin/app/models/match.rb]: No need to name it and return it.

File: components/react/src/App.js (hunk 4 above)
Comment (pauljickling) [reply components/react/src/App.js]: Agreed, this should be wrapped in a `Promise`. We can do that using `async/await` or just create a `Promise` object.

File: components/react/src/App.js [hunk 8]
```diff
@@ -1,56 +1,58 @@
...
```
Comment (tsell) [inline components/react/src/App.js]: localhost:5455 won't work in a deployed environment, should be a configurable value.

File: components/react/src/App.js [hunk 9]
```diff
@@ -1,56 +1,58 @@
...
-    } else if (!isLoaded) {
-      return <div className="App"><Header />Loading...</div>;
-    } else {
-      return (
-        <div className="App">
-          <Header />
-          <SelectorMediator {...this.state} />
-          <ProjectList {...this.state} />
+  const generateMatch = () => fetch(
+    'http://localhost:5455/matches.json',
```
Comment (tsell) [inline components/react/src/App.js]: Same comment, localhost:5455 only works locally.

Comment (rickpr) [reply components/react/src/App.js]: Fixed, now it is an environment variable called `API_URL`.

File: components/react/src/App.js (hunk 4 above)
Comment (rickpr) [reply components/react/src/App.js]: Fixed, now this uses `async/await` to make the fetch.

File: components/react/src/App.js [hunk 10]
```diff
@@ -1,56 +1,64 @@
...
-  constructor(props) {
-    super(props);
-    this.state = {
-      isLoaded: false,
-      projects: [],
-    };
-  }
-
-  componentDidMount() {
-    this.request('http://localhost:5455/projects.json', 'projects')
```
Comment (tsell) [inline components/react/src/App.js]: localhost bad, ApiUrl variable good

**Pull request merged**